src/main/java/ro/developmentfactory/arrayProblem/Main.java

```diff
@@ -0,0 +1,40 @@
+package ro.developmentfactory.arrayProblem;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+public class Main {
+    public static void main(String[] args) {
+
+        ExecutorService executor = Executors.newFixedThreadPool(5);
+        int numberOfTasks = 5;
+        List<IntermediateSuma> tasks = new ArrayList<>();
+
+        for (int i = 0; i < numberOfTasks; i++) {
+            IntermediateSuma problem = new IntermediateSuma();
```

Comment:
your input array is in `IntermediateSuma` class, so what you are doing in `Main` is:
* calculating the sum of { 23, 45, 67, 89, 12, 34, 56, 78, 90, 1 } 5 times
* overwriting the result of `totalSum` 5 times

you are not distributing subparts of the input array to 5 different threads